src-tauri/src/nostr.rs

```diff
@@ -80,3 +98,65 @@ pub async fn get_contacts() -> Vec<Contact> {
         .await
         .unwrap()
 }
+
+#[tauri::command]
+pub async fn get_legacy_chats(pubkey: String) -> HashMap<String, Conversation> {
+    let client = get_client().expect("Couldn't get the nostr client");
+    let current_pubkey = PublicKey::from_hex(&pubkey).unwrap();
+    let filter = Filter::new()
+        .kind(Kind::EncryptedDirectMessage)
+        .authors(vec![current_pubkey]);
+
+    let filter2 = Filter::new()
+        .kind(Kind::EncryptedDirectMessage)
+        .pubkeys(vec![current_pubkey]);
+    
+    let events = client.get_events_of(vec![filter, filter2], EventSource::both(Some(DEFAULT_TIMEOUT))).await.unwrap();
+    
+    let mut chats: HashMap<String, Conversation> = HashMap::new();
+    let signer = client.signer().await.unwrap();
+    let signer_pubkey = signer.public_key().await.unwrap();
+
+    for event in events {
+        let (other_party_pubkey, decrypt_pubkey) = if event.author() == signer_pubkey {
+            let other_pubkey = PublicKey::parse(event.get_tag_content(TagKind::SingleLetter(SingleLetterTag::lowercase(Alphabet::P)))
```

Comment:
So much ceremony lol

Reply:
Seriously. This took me like an hour to work out. I was furious. 😂